clients/src/main/java/org/apache/kafka/common/requests/OffsetFetchRequest.java

```diff
@@ -207,47 +159,16 @@ public OffsetFetchResponse getErrorResponse(int throttleTimeMs, Throwable e) {
         return getErrorResponse(throttleTimeMs, Errors.forException(e));
     }
 
-    public String groupId() {
-        return groupId;
-    }
-
-    public List<TopicPartition> partitions() {
-        return partitions;
-    }
-
     public static OffsetFetchRequest parse(ByteBuffer buffer, short version) {
         return new OffsetFetchRequest(ApiKeys.OFFSET_FETCH.parseRequest(version, buffer), version);
     }
 
     public boolean isAllPartitions() {
-        return partitions == null;
+        return data.topics() == Builder.ALL_TOPIC_PARTITIONS;
```

Comment:
nit: since we need `ALL_TOPIC_PARTITIONS` outside of Builder, perhaps we should pull it up to `OffsetFetchRequest`?